### PR TITLE
Portablectl patches for to support --now --enable and --no-block flags

### PR DESCRIPTION
**Pull request by chalamalasetty**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
    Portablectl is missing --now --enable and --no-block flags which are required in CBL-Mariner.
    These flags are require for systemd generators. Systemd 239 version is stable hence back porting required patches.
    These patches can be removed once we upgrade systemd version to latest. 
###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**YES**
NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/...

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/systemd/systemd.spec [hunk 1]
```diff
@@ -36,6 +36,15 @@ Patch15:          https://github.com/systemd/systemd/commit/8f6b442a78d0b485f044
 # Furthermore, strict mode DoT is not supported before v243.
 Patch16:          CVE-2018-21029.nopatch
 
+#Portablectl patches for --now --enable and --no-block flags support
+Patch100:          100-portabled-allow-to-detach-an-image-with-a-unit-in-li.patch
```
Comment (MateuszMalisz) [inline SPECS/systemd/systemd.spec:40]: Why do these start with 100 instead of continuing from 16?

File: SPECS/systemd/systemd.spec [hunk 2]
```diff
@@ -108,6 +117,15 @@ EOF
 %patch14 -p1
 %patch15 -p1
 
+# Portablectl patches
+%patch100 -p1
```
Comment (MateuszMalisz) [inline SPECS/systemd/systemd.spec:121]: I suggest using `%autosetup` instead and removing all `%patch` directives.

File: SPECS/systemd/systemd.spec (hunk 1 above)
Comment (dmcilvaney) [reply SPECS/systemd/systemd.spec:40]: I've actually done this as well, its very convenient to split patches into groups (0-99 CVEs, 100-199 features, 200-299 extra stuff, etc). This means you can add an extra CVE patch onto the sources when they are as close to upstream as possible, but don't have to re-number all subsequent patches.

File: toolkit/resources/manifests/package/toolchain_aarch64.txt [hunk 3]
```diff
@@ -352,13 +352,13 @@ sqlite-devel-3.32.3-1.cm1.aarch64.rpm
 sqlite-libs-3.32.3-1.cm1.aarch64.rpm
 swig-3.0.12-4.cm1.aarch64.rpm
 swig-debuginfo-3.0.12-4.cm1.aarch64.rpm
-systemd-239-31.cm1.aarch64.rpm
+systemd-239-32.cm1.aarch64.rpm
 systemd-bootstrap-239-29.cm1.aarch64.rpm
```
Comment (dmcilvaney) [inline toolkit/resources/manifests/package/toolchain_aarch64.txt:356]: Do these changes need to be applied to the bootstrap version as well?

**Review by dmcilvaney — APPROVED** [review]

File: toolkit/resources/manifests/package/toolchain_aarch64.txt (hunk 3 above)
Comment (chalamalasetty) [reply toolkit/resources/manifests/package/toolchain_aarch64.txt:356]: AFAIK, not all changes needed in bootstarp. I think this change is also not needed. It is to add additional flags for portablectl.

File: SPECS/systemd/systemd.spec (hunk 2 above)
Comment (chalamalasetty) [reply SPECS/systemd/systemd.spec:121]: Systemd SPEC already using %patches hence using the same. To move to %autosetup we need more validation to confirm all patches applied correctly. Hence going with %patch.

Comment (PawelWMS) [reply SPECS/systemd/systemd.spec:121]: What validation do you have in mind? If a patch doesn't apply correctly the package will fail to build, so any breaks should be quickly visible.

**Comment by PawelWMS** [comment]
Please fill in the detailes asked for in the PR's description template. We're striving to provide a uniform and detailed description of each change, so that it's easier to track and understand the changes to both internal, as well as external viewers.

**Review by PawelWMS — CHANGES_REQUESTED** [review]
:clock1:

File: SPECS/systemd/systemd.spec (hunk 2 above)
Comment (WSLUser) [reply SPECS/systemd/systemd.spec:121]: Clear Linux also does this and autospec successfully applies the patches. So does this really need to change?

**Review by jslobodzian — APPROVED** [review]
:shipit:

**Pull request merged**